File: app/src/main/java/com/timsimonhughes/atlas/model/POTD.java
package com.timsimonhughes.atlas.model;

import android.os.Parcel;
import android.os.Parcelable;

public class POTD implements Parcelable {

    private String date;
    private String explanation;
    private String hdurl;
    private String media_type;
    private String service_version;
    private String title;
    private String url;

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getExplanation() {
        return explanation;
    }

    public void setExplanation(String explanation) {
        this.explanation = explanation;
    }

    public String getHdurl() {
        return hdurl;
    }

    public void setHdurl(String hdurl) {
        this.hdurl = hdurl;
    }

    public String getMedia_type() {
        return media_type;
    }

    public void setMedia_type(String media_type) {
        this.media_type = media_type;
    }

    public String getService_version() {
        return service_version;
    }

    public void setService_version(String service_version) {
        this.service_version = service_version;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public POTD(Parcel in) {
        date = in.readString();
        explanation = in.readString();
        hdurl = in.readString();
        media_type = in.readString();
        service_version = in.readString();
        title = in.readString();
        url = in.readString();
    }

    public static final Creator<POTD> CREATOR = new Creator<POTD>() {
        @Override
        public POTD createFromParcel(Parcel in) {
            return new POTD(in);
        }

        @Override
        public POTD[] newArray(int size) {
            return new POTD[size];
        }
    };

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(date);
        dest.writeString(explanation);
        dest.writeString(hdurl);
        dest.writeString(media_type);
        dest.writeString(service_version);
        dest.writeString(title);
        dest.writeString(url);
    }
}

File: app/src/main/java/com/timsimonhughes/atlas/model/POTD.kt
package com.timsimonhughes.atlas.model

import android.os.Parcel
import android.os.Parcelable
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class POTD(
        @field:Json(name = "date")
        val date: String?,
        @field:Json(name = "explanation")
        val explanation: String?,
        @field:Json(name = "hdurl")
        val hrUrl: String?,
        @field:Json(name = "media_type")
        val mediaType: String?,
        @field:Json(name = "service_version")
        val serviceVersion: String?,
        @field:Json(name = "title")
        val title: String?,
        @field:Json(name = "url")
        val url: String?
) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(date)
        parcel.writeString(explanation)
        parcel.writeString(hrUrl)
        parcel.writeString(mediaType)
        parcel.writeString(serviceVersion)
        parcel.writeString(title)
        parcel.writeString(url)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<POTD> {
        override fun createFromParcel(parcel: Parcel): POTD {
            return POTD(parcel)
        }

        override fun newArray(size: Int): Array<POTD?> {
            return arrayOfNulls(size)
        }
    }
}